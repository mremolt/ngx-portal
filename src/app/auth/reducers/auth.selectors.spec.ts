import { HttpErrorResponse } from '@angular/common/http';

import {
  AuthState,
  selectAuth,
  selectAuthenticated,
  selectError,
  selectLoading,
  selectToken,
} from './auth.selectors';

describe('auth selectors', () => {
  const state: AuthState = {
    auth: {
      loading: false,
      token: { accessToken: 'entry' },
      error: new HttpErrorResponse({ error: 'ARGH' }),
    },
    router: {} as any,
  };

  describe('selectAuth', () => {
    it('returns auth state', () => {
      expect(selectAuth(state)).toMatchSnapshot();
    });
  });

  describe('selectLoading', () => {
    it('returns sub state', () => {
      expect(selectLoading(state)).toMatchSnapshot();
    });
  });

  describe('selectToken', () => {
    it('returns sub state', () => {
      expect(selectToken(state)).toMatchSnapshot();
    });
  });

  describe('selectError', () => {
    it('returns sub state', () => {
      expect(selectError(state)).toMatchSnapshot();
    });
  });

  describe('selectAuthenticated', () => {
    it('returns sub state', () => {
      expect(selectAuthenticated(state)).toMatchSnapshot();
    });
  });
});
