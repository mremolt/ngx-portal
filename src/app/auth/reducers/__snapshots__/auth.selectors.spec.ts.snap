// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`auth selectors selectAuth returns auth state 1`] = `
Object {
  "error": HttpErrorResponse {
    "error": "ARGH",
    "headers": HttpHeaders {
      "headers": Map {},
      "lazyUpdate": null,
      "normalizedNames": Map {},
    },
    "message": "Http failure response for (unknown url): undefined undefined",
    "name": "HttpErrorResponse",
    "ok": false,
    "status": 0,
    "statusText": "Unknown Error",
    "url": null,
  },
  "loading": false,
  "token": Object {
    "accessToken": "entry",
  },
}
`;

exports[`auth selectors selectAuthenticated returns sub state 1`] = `true`;

exports[`auth selectors selectError returns sub state 1`] = `
HttpErrorResponse {
  "error": "ARGH",
  "headers": HttpHeaders {
    "headers": Map {},
    "lazyUpdate": null,
    "normalizedNames": Map {},
  },
  "message": "Http failure response for (unknown url): undefined undefined",
  "name": "HttpErrorResponse",
  "ok": false,
  "status": 0,
  "statusText": "Unknown Error",
  "url": null,
}
`;

exports[`auth selectors selectLoading returns sub state 1`] = `false`;

exports[`auth selectors selectToken returns sub state 1`] = `
Object {
  "accessToken": "entry",
}
`;
